python/ray/data/_internal/execution/backpressure_policy/downstream_output_capacity_backpressure_policy.py

```diff
@@ -0,0 +1,179 @@
+# File: python/ray/data/_internal/execution/backpressure_policy/downstream_capacity_output_backpressure_policy.py
+
+import logging
+from typing import TYPE_CHECKING, Optional
+
+from .backpressure_policy import BackpressurePolicy
+from ray.data._internal.execution.operators.actor_pool_map_operator import (
+    ActorPoolMapOperator,
+)
+from ray.data.context import DataContext
+
+if TYPE_CHECKING:
+    from ray.data._internal.execution.interfaces.physical_operator import (
+        PhysicalOperator,
+    )
+    from ray.data._internal.execution.resource_manager import ResourceManager
+    from ray.data._internal.execution.streaming_executor_state import Topology
+
+logger = logging.getLogger(__name__)
+
+
+class DownstreamOutputCapacityBackpressurePolicy(BackpressurePolicy):
```

Comment:
This class conceptually overlaps with the existing `DownstreamCapacityBackpressurePolicy`. Rather than introducing a new policy, it might be simpler if we extend `DownstreamCapacityBackpressurePolicy` and implement `max_task_output_bytes_to_read`.

We might also be able to reuse some of the code to compute downstream operator capacity.